ntp-proto/src/algorithm/kalman/select.rs

```diff
@@ -38,30 +43,44 @@ pub(super) fn select<Index: Copy>(
 
     bounds.sort_by(|a, b| a.0.total_cmp(&b.0));
 
-    let mut max: usize = 0;
-    let mut maxt: f64 = 0.0;
+    let mut maxlow: usize = 0;
+    let mut maxhigh: usize = 0;
+    let mut maxtlow: f64 = 0.0;
+    let mut maxthigh: f64 = 0.0;
     let mut cur: usize = 0;
 
     for (time, boundtype) in bounds.iter() {
         match boundtype {
-            BoundType::Start => cur += 1,
-            BoundType::End => cur -= 1,
-        }
-        if cur > max {
-            max = cur;
-            maxt = *time;
+            BoundType::Start => {
```

Comment:
I think we should probably update the comment on top of this function (and/or maybe add a few inline comments), to explain why we now look for both the lower and higher bound.